### ISSUE TITLE
Style/Encoding AutoCorrect with EnforcedStyle: when_needed goes into an infinite loop

### ISSUE THREAD

**Issue by arantir** [bbatsov/rubocop#2631]
rubocop version: master at 9732d37558
`0.35.1 (using Parser 2.3.0.pre.6, running on ruby 2.3.0 x86_64-linux)`

.rubocop.yml:
```yml
AllCops:
  TargetRubyVersion: 2.3

Style/Encoding:
  Enabled: true
  EnforcedStyle: when_needed
```

My actions:
```
$ bundle exec rubocop 
file.rb:1:1: C: Unnecessary utf-8 encoding comment.
# encoding: utf-8
^
```

```
$ bundle exec rubocop --auto-correct
```

Expected behaviour: line `# encoding: utf-8` is deleted from all files.
Actual behaviour: infinite addition of `# encoding: utf-8` lines to the first encountered file.

**Comment by arantir** [comment]
Just a little thought: for `TargetRubyVersion` >= 2.0 this cop should always report encoding comment as unnecessary regardless of `EnforcedStyle`.

**Comment by pasha** [comment]
Having similar issue.
using latest version 0.37.2
ruby 2.2.3p173

```yml
Style/Encoding:
  Description: 'Use UTF-8 as the source file encoding.'
  StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#utf-8'
  EnforcedStyle: when_needed
  Enabled: true
```
